server/services/prison/PrisonRegisterService.ts

```diff
@@ -1,29 +1,37 @@
-import RestClient from '../../data/restClient'
 import config from '../../config'
+import PrisonRegisterStore from '../../data/cache/PrisonRegisterStore'
+import RestClient from '../../data/restClient'
 import { PrisonDto } from '../../@types/prisonRegisterApiClientTypes'
-
-export interface Prison {
-  id: string
-  name: string
-}
+import { Prison } from './PrisonTypes'
 
 export default class PrisonRegisterService {
+  constructor(private readonly prisonRegisterStore: PrisonRegisterStore) {}
+
   private static restClient(): RestClient {
     return new RestClient('Prison Register API Client', config.apis.prisonRegister)
   }
 
   async getActivePrisons(): Promise<Array<Prison>> {
-    return PrisonRegisterService.restClient()
-      .get({ path: '/prisons' })
-      .then(data =>
-        (data as Array<PrisonDto>)
-          .filter(prison => prison.active === true)
-          .map(prisonDto => {
-            return {
-              id: prisonDto.prisonId,
-              name: prisonDto.prisonName,
-            }
-          })
-      )
+    try {
```

Comment:
Not sure how comfortable I feel about this and the asynchronous nature of it.
If it were java you'd make it `synchronized` in some way. I know that javascript is inherently single threaded and that node uses an event loop to prevent the single thread blocking.
I think its the concept of the event loop that is making me stop and think about this .... there is definitely scope for 2 requests to hit this method and for them both to get nothing back from redis and for them both to then request the data from the prison register and for them both to cache it.
In reality I know this will be rare (if ever) and if it does happen it wont do any harm. It just feels wrong ..... ?

Reply:
No harm no foul. Complicated synchronize logic doesn't seem worth the effort... and will make this even more complicated.